ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-sync/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/sync/syncer/DgcRulesClient.java

```diff
@@ -42,23 +50,83 @@ public DgcRulesClient(String dgcBaseUrl, RestTemplate dgcRT, SigningClient signi
     }
 
     /**
-     * downloads rules for all countries
+     * downloads rules for Switzerland
      *
-     * @return rules per country
+     * @return rules for Switezrland
      */
-    public Map<String, String> download() { // TODO
+    public Map<String, JsonNode> download() {
         logger.info("[DgcRulesClient] Downloading rules");
+        Map<String, JsonNode> rules = new HashMap<>();
+        ResponseEntity<String> response = this.dgcRT.exchange(RequestEntity.get(dgcBaseUrl+String.format(DOWNLOAD_PATH, "CH")).headers(CmsUtil.createCmsTextUploadHeaders()).build(),
+                String.class);
+        try {
+            new ObjectMapper().readTree(response.getBody()).fields().forEachRemaining( field -> {
+                rules.put(field.getKey(), field.getValue());
+            });
+        } catch (JsonProcessingException e) {
+            logger.error("[DgcRulesClient] Failed to deserialize downloaded rules", e);
+        }
+
+        logger.info("[DgcRulesClient] downloaded {} rules", rules.size());
+        return rules;
+    }
+
+    /**
+     * deletes the rules with the given IDs
+     * @param identifiers IDs of the rules to delete
+     * @return successfully deleted rule IDs
+     */
+    public RulesSyncResult deleteRules(Collection<String> identifiers){
+        logger.info("Deleting {} rules", identifiers);
+        List<String> deletedRuleIds = new ArrayList<>();
+        List<String> failedRuleIds = new ArrayList<>();
+        identifiers.forEach(ruleId -> {
+                try {
```

Comment:
Unify those try and catches since there are never actions executed after the catch block.

Reply:
Could you explain what exactly you want to merge? I'm not sure how we would do that without losing information about where things failed. Also, should we change it in the `upload` function too then? It's the same structure as here.

Reply:
Something along the lines of: 
```java
try {
   someCodeWhichCouldFail();
   otherCodeWhichCouldFail();
} catch (SomeCodeException ex) {
  handleSomeCodeException();
} catch (OtherCodeException ex) {
  handleOtherCodeException();
} catch (Exception ex) {
  someGenericCatch();
}
```

The http status code exception is already different, the sign function is controlled by us, so we could return a typed exception with e.g. an inner exception and then have a catch all for the rest.

Yes would probably make sense to change that in the upload function as well, I was just browsing through the code smells (and sonarcloud only highlights new stuff)